code.js

```diff
@@ -0,0 +1,56 @@
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+// Initialize the S3 client outside the handler for reuse
+const s3Client = new S3Client();
+
+/**
+ * Lambda handler for processing orders and storing receipts in S3.
+ * @param {Object} event - Input event containing order details
+ * @param {string} event.order_id - The unique identifier for the order
+ * @param {number} event.amount - The order amount
+ * @param {string} event.item - The item purchased
+ * @returns {Promise<string>} Success message
+ */
+export const handler = async(event) => {
+    try {
+        // Access environment variables
+        const bucketName = process.env.RECEIPT_BUCKET;
+        if (!bucketName) {
+            throw new Error('RECEIPT_BUCKET environment variable is not set');
+        }
+
+        // Create the receipt content and key destination
+        const receiptContent = `OrderID: ${event.order_id}\nAmount: $${event.amount.toFixed(2)}\nItem: ${event.item}`;
```

Comment:
> [!WARNING]
> The code doesn't sanitize user inputs before using them in the receipt content, which could lead to injection attacks if the input contains special characters or formatting that could be maliciously crafted.

> [!TIP]
> Sanitize user inputs to prevent potential injection attacks in the receipt content.

```javascript
// Sanitize inputs to prevent injection
const sanitizedOrderId = String(event.order_id).replace(/[\n\r]/g, '');
const sanitizedItem = String(event.item).replace(/[\n\r]/g, '');

const receiptContent = `OrderID: ${sanitizedOrderId}\nAmount: $${event.amount.toFixed(2)}\nItem: ${sanitizedItem}`;
```